Add --stimulus flag to component generator

websebdev opened the pull request:
### Summary
Add `--stimulus` flag to the component generator.
This will create a basic Stimulus controller alongside the component.

Example:
```
$ rails generate component User --stimulus
      create  app/components/user_component.rb
      invoke  rspec
      create    spec/components/user_component_spec.rb
      invoke  stimulus
      create    app/components/user_component_controller.js
      invoke  erb
      create    app/components/user_component.html.erb
```

Closes #627

joelhawksley reviewed (commented):
Looks great! Might we add a global option for this setting? We should also add documentation for this new functionality ❤️

websebdev commented:
@joelhawksley Done.
I added the global option `config.view_component.generate_stimulus_controller`. When set to `true`, the generator will automatically generate the stimulus controller. I'm not sure if that's what you had in mind though.

Also added documentation for both the new config option and the `---stimulus` flag.

Commit: https://github.com/github/view_component/pull/1030/commits/9346c714d6328e542218899993d3bffbd3081b9f

joelhawksley commented on docs/CHANGELOG.md:
```diff
@@ -8,6 +8,7 @@ title: Changelog
 ## main
 
 * Add `--stimulus` flag to the component generator. Generates a Stimulus controller alongside the component.
+* Add config option `config.view_component.generate_stimulus_controller` to always generate a Stimulus controller.
```
@Spone does this config attribute naming work for you?

Spone replied:
Yep 👍👍

joelhawksley reviewed (approved):
Thank you for your contribution! ❤️

Spone commented:
@joelhawksley I noticed the documentation for the new `generate_stimulus_controller` option is missing from the documentation: https://viewcomponent.org/api.html

Got any idea why?
